code/modules/mob/living/carbon/xenomorph/Xenomorph.dm

```diff
@@ -347,157 +347,158 @@
 		mob_flags |= SQUEEZE_UNDER_VEHICLES
 
+	// More setup stuff for names, abilities etc
+	update_icon_source()
 	generate_name()
+	add_inherent_verbs()
+	add_abilities()
+	create_reagents(100)
+	regenerate_icons()
+	toggle_xeno_mobhud() //This is a verb, but fuck it, it just werks
+	toggle_xeno_hostilehud()
+	recalculate_everything()
 
-	if(isqueen(src))
-		SStracking.set_leader("hive_[hivenumber]", src)
+	//Begin SStracking
 	SStracking.start_tracking("hive_[hivenumber]", src)
 
 	. = ..()
+
+	GLOB.living_xeno_list += src
+	GLOB.xeno_mob_list += src
+
 	//WO GAMEMODE
 	if(SSticker?.mode?.hardcore)
 		hardcore = 1 //Prevents healing and queen evolution
 	time_of_birth = world.time
 
-	add_inherent_verbs()
-	add_abilities()
-	recalculate_actions()
-
+	//Minimap
 	if(z)
 		INVOKE_NEXT_TICK(src, PROC_REF(add_minimap_marker))
 
+	//Sight
 	sight |= SEE_MOBS
 	see_invisible = SEE_INVISIBLE_LIVING
 	see_in_dark = 12
+
 	if(client)
 		set_lighting_alpha_from_prefs(client)
 	else
 		lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 
-	if(caste && caste.spit_types && caste.spit_types.len)
-		ammo = GLOB.ammo_list[caste.spit_types[1]]
-
-	create_reagents(100)
-
-	GLOB.living_xeno_list += src
-	GLOB.xeno_mob_list += src
-
-	if(caste && caste.adjust_size_x != 1)
-		var/matrix/M = matrix()
-		M.Scale(caste.adjust_size_x, caste.adjust_size_y)
-		apply_transform(M)
-
-	if(caste)
-		behavior_delegate = new caste.behavior_delegate_type()
-		behavior_delegate.bound_xeno = src
-		behavior_delegate.add_to_xeno()
-		resin_build_order = caste.resin_build_order
-	else
-		CRASH("Xenomorph [src] has no caste datum! Tell the devs!")
-
-	regenerate_icons()
-	toggle_xeno_mobhud() //This is a verb, but fuck it, it just werks
-	toggle_xeno_hostilehud()
-
-	if(oldXeno)
-		a_intent_change(oldXeno.a_intent)//Keep intent
-
-		if(oldXeno.layer == XENO_HIDING_LAYER)
-			//We are hiding, let's keep hiding if we can!
-			for(var/datum/action/xeno_action/onclick/xenohide/hide in actions)
-				if(istype(hide))
-					layer = XENO_HIDING_LAYER
-					hide.button.icon_state = "template_active"
-
-		for(var/obj/item/W in oldXeno.contents) //Drop stuff
-			oldXeno.drop_inv_item_on_ground(W)
-
-		oldXeno.empty_gut()
-
-		if(IS_XENO_LEADER(oldXeno))
-			hive.replace_hive_leader(oldXeno, src)
-
 	// Only handle free slots if the xeno is not in tdome
-	if(!is_admin_level(z))
+	if(hive && !is_admin_level(z))
 		var/selected_caste = GLOB.xeno_datum_list[caste_type]?.type
 		hive.used_slots[selected_caste]++
 
+	//Statistics
+	var/area/current_area = get_area(src)
+	if(current_area && current_area.statistic_exempt)
+		statistic_exempt = TRUE
 	if(round_statistics && !statistic_exempt)
 		round_statistics.track_new_participant(faction, 1)
-	generate_name()
 
 	// This can happen if a xeno gets made before the game starts
 	if (hive && hive.hive_ui)
 		hive.hive_ui.update_all_xeno_data()
 
-	job = caste.caste_type // Used for tracking the caste playtime
 	Decorate()
```

Comment:
why is this here, i'm gonna go insane

Reply:
Xenomorphs off to their jobs 🕴

Reply:
no i meant the Decorate, this is already handled at atom level 😩

Reply:
Oh I see. Yea I actually don't know this is one of those things I didn't touch yet.